### ISSUE TITLE
Commit 02e474d broke our client

### ISSUE THREAD

**Issue by doublemarked** [swagger-api/swagger-js#272]
Hello! A script of ours using swagger-js suddenly stopped working. Our client code is doing nothing special - a call to a method that takes a POST body payload:
```javascript
client.Users.login({body: {
      username: CONF.username,
      password: CONF.password
    }
  });
```
I believe this is a standard use of the library, yes? It's very similar to the sample code involving ``addPet``.

I spent some time digging into the recent commits to the project and have determined that the following code from commit 02e474d (lines starting 2839) is the culprit,

```javascript
obj.body = JSON.stringify(obj.body);
 if(obj.body.type && obj.body.type !== 'formData')
   obj.body = JSON.stringify(obj.body);
 else {
   obj.contentType = false;
   obj.processData = false;
   // delete obj.cache;
  delete obj.headers['Content-Type'];
  }
}
```

I do not know the codebase, but this block of code seems twisted in several ways and I'm not sure I can make sense of it. What I do know - if I short circuit that ``body.type`` conditional, my code starts working. Otherwise the Content-Type gets deleted and my server gets bad data.

Please advise?

**Comment by fehguy** [comment]
yes @doublemarked this is a new bug.  Working on a fix and tests so this won't happen again.

**Comment by fehguy** [comment]
fixed in v2.1.10-M1

**Comment by doublemarked** [comment]
Thank you! :)
